pkg/cosign/tuf/client_test.go

```diff
@@ -308,6 +310,84 @@ func TestGetTargetsByMeta(t *testing.T) {
 	}
 }
 
+func makeMapFS(repo string) (fs fstest.MapFS) {
+	fs = make(fstest.MapFS)
+	_ = filepath.Walk(repo,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			rel, _ := filepath.Rel(repo, path)
+			if info.IsDir() {
+				fs[filepath.FromSlash(filepath.Join("repository", rel))] = &fstest.MapFile{Mode: os.ModeDir}
+			} else {
+				b, _ := os.ReadFile(path)
+				fs[filepath.FromSlash(filepath.Join("repository", rel))] = &fstest.MapFile{Data: b}
+			}
+			return nil
+		})
+	return
+}
+
+// Regression test for failure to locate newly added/renamed targets from a TUF update.
+// Previously, the embedded or in-memory targetImpl did not fetch any updated targets.
+func TestUpdatedTargetNamesEmbedded(t *testing.T) {
+	td := t.TempDir()
+	// Set the TUF_ROOT so we don't interact with other tests and local TUF roots.
+	t.Setenv("TUF_ROOT", td)
+
+	origEmbedded := GetEmbedded
+	origDefaultRemote := GetRemoteRoot
+	defer func() {
+		GetEmbedded = origEmbedded
+		GetRemoteRoot = origDefaultRemote
+	}()
+
+	// Create an "expired" embedded repository.
+	ctx := context.Background()
+	store, r := newTufCustomRepo(t, td, "foo")
+	repository := filepath.FromSlash(filepath.Join(td, "repository"))
+	mapfs := makeMapFS(repository)
+	GetEmbedded = func() fs.FS { return mapfs }
+
+	oldIsExpired := verify.IsExpired
+	isExpiredTimestamp = func(metadata []byte) bool {
+		m, _ := store.GetMeta()
+		timestampExpires, _ := getExpiration(m["timestamp.json"])
+		metadataExpires, _ := getExpiration(metadata)
+		return metadataExpires.Sub(*timestampExpires) <= 0
+	}
+
+	// Serve an updated remote repository with a new target.
+	addNewCustomTarget(t, td, r, "newdata")
+	s := httptest.NewServer(http.FileServer(http.Dir(repository)))
+	defer s.Close()
+	GetRemoteRoot = func() string { return s.URL }
+
+	// Initialize.
+	tufObj, err := NewFromEnv(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tufObj.Close()
+
+	// Try to retrieve the newly added target.
+	targets, err := tufObj.GetTargetsByMeta(Fulcio, []string{"fooNoCustom.txt"})
```

Comment:
I specifically want to test that it can fetch a new target WITHOUT initializing the TUF obj first and potentially having a cache. (this is what failed before). What I can do is assert that the new target was not in the embedded metadata, but I don't want to create a `tufObj` at this point.

Reply:
Oops, that's a good point. Maybe a comment to that effect  (the top-level one for the test)? That says "we wanna fetch before initializing the TUF object"